### ISSUE TITLE
AsyncContextManager fasly raises inherit-non-class

### ISSUE THREAD

**Issue by JulienPalard** [PyCQA/pylint#2472]
### Steps to reproduce

```py
from typing import AsyncContextManager

class Foo(AsyncContextManager):
    ...
```

### Current behavior

`pylint: Inheriting 'AsyncContextManager', which is not a class.  ("inherit-non-class") [E0239]`

### Expected behavior

No error as it's a class:
```
>>> import typing
>>> type(typing.AsyncContextManager)
<class 'typing._GenericAlias'>
```

### pylint --version output

```
pylint 2.1.1
astroid 2.0.4
Python 3.7.0 (default, Jun 27 2018, 14:40:03) 
[GCC 8.1.0]
```

I bet it's however hard to spot that it's a class as the typing module is a bit convoluted.

**Comment by cdce8p** [comment]
This seems to be fixed in the latest version of pylint.
If you still experience issues, please report back and I'll reopen it.
